scipy/stats/tests/test_stats.py

```diff
@@ -541,6 +562,54 @@ def test_spearmanr_result_attributes(self):
         attributes = ('correlation', 'pvalue')
         check_named_results(res, attributes)
 
+    def test_1d_vs_2d(self):
+        x1 = [1, 2, 3, 4, 5, 6]
+        x2 = [1, 2, 3, 4, 6, 5]
+        res1 = stats.spearmanr(x1, x2)
+        res2 = stats.spearmanr(np.asarray([x1, x2]).T)
+        assert_allclose(res1, res2)
+
+    def test_1d_vs_2d_nans(self):
+        # Now the same with NaNs present.  Regression test for gh-9103.
+        for nan_policy in ['propagate', 'omit']:
+            x1 = [1, np.nan, 3, 4, 5, 6]
+            x2 = [1, 2, 3, 4, 6, np.nan]
+            res1 = stats.spearmanr(x1, x2)
+            res2 = stats.spearmanr(np.asarray([x1, x2]).T)
```

Comment:
is the iterator variable `nan_policy` somehow getting automagically fed in to `spearmanr` here? I think you need to feed it directly to the kwarg with the same name, right?

Might also be worth parametrizing w/ pytest for the policies while you're at it.

Reply:
hmm, sloppy. yes you are correct

Reply:
I'd prefer not to use `parametrize` for such simple len-2 for-loops, that makes the code longer and less readable for little gain.